src/interfaces/protocols/types.ts

```diff
@@ -10,16 +10,19 @@ export type ProtocolsConfigureDescriptor = {
 };
 
 export type ProtocolDefinition = {
-  recordDefinitions: ProtocolRecordDefinition[];
+  types: ProtocolTypes;
   records: {
     [key: string]: ProtocolRuleSet;
-  };
+  }
 };
 
-export type ProtocolRecordDefinition = {
-  id: string,
+export type ProtocolType = {
   schema?: string,
-  dataFormats?: string[],
+  dataFormats: string[],
+};
+
+export type ProtocolTypes = {
+  [key: string]: ProtocolType;
```

Comment:
Discussion: Given the merge of https://github.com/TBD54566975/dwn-sdk-js/commit/6bc148c72eaee2e8fd35b63753a764bad98661e5, do you think we should use `Record<T, U>` consistently?

Personally I much prefer the `[key:string]: T` style because `Record` is a first class concept in DWN and can easily be confused with `Record<T, U>` for people not familiar with TypeScript, but I never got around to review the PR.

If you agree we can revert the `Record<T, U>` changes in separate PR.

Reply:
Let's keep `[key: string]`. I like your point about `Record` already having meaning in dwn-sdk. In other contexts I usually prefer `Record` as I find it more readable.